app/src/main/java/com/example/mytestproject/ui/weatherData/todayWeather/CurrentWeatherFragment.kt

```diff
@@ -60,24 +60,26 @@ class CurrentWeatherFragment : Fragment() {
         btn_retry.setOnClickListener {
             currentWeatherViewModel.onRetry()
         }
+
+        refreshingWeather()
     }
 
     private fun setupWeatherData() {
         currentWeatherViewModel.viewState.observe(viewLifecycleOwner, Observer {
             when (it) {
```

Comment:
Let's simplify it a bit. We can move all negative strings outside `when`. 

I mean strings:
group_temp_abbreviation.visibility = View.GONE
group_error_views.visibility = View.GONE
refresh_layout.isRefreshing = false

They will be called for every state. It's totally fine. 

Inside `when` we keep only positive strings:  
refresh_layout.isRefreshing = true
group_error_views.visibility = View.VISIBLE
group_temp_abbreviation.visibility = View.VISIBLE

This trick will help to get rid of some duplicating and confusing code.
The same is actual for screens 3days and 10days.

Reply:
Fixed